mondrian/src/main/resources/mondrian.xsd

```diff
@@ -3,7 +3,12 @@
 
   <!-- region: processed elements -->
 
-  <xs:element name="Schema" type="SchemaType"/>
+  <xs:element name="Schema" type="SchemaType">
+    <xs:unique name="UniqueCubeName">
```

Comment:
Shouldn't we use a xs:key instead of a xs:unique? 
A quick google search turned out the following:
> - xsd:unique does not require the field to exist for the selector. If the field exists, its value should be unique.
> - xsd:key requires the field to exist for all selector results. Every element in the selector should have a key.
I know that we are requiring the attribute in the [CubeType](https://github.com/pentaho/mondrian/pull/1418/files#diff-dbcec313a1b711674da59e38c3d968361d664e0ec44952a6d8d07032b1afc0e8R453) but still...

Reply:
I'll test with `xs:key`

Reply:
Updated with `xs:key`